app/services/attribute_asserter.rb

```diff
@@ -15,10 +13,10 @@ class AttributeAsserter
     :phone
   ].freeze
 
-  def initialize(user, service_provider, authn_request)
-    self.user = user
-    self.service_provider = service_provider
-    self.authn_request = authn_request
+  URI_PATTERN = Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF
+
+  def initialize(options)
```

Comment:
I agree, am a big fan of named arguments, then we get bonuses like errors when there are extra/typoed arguments

Since we know we're on a recent version of Ruby and AFAIK are not planning on switching platforms (ex JRuby) I think there's a huge upside to named arguments